src/main/java/Duke.java

```diff
@@ -1,10 +1,135 @@
+        }
+        System.out.println(lineBreaker);
+    }
+
+    public void handleError() {
+        System.out.println(errorMessage);
+    }
+
+    public void printTotalTasks(int currentTaskNum) {
+        if (currentTaskNum == 1) {
+            System.out.println("Now you have " + currentTaskNum + " task in the list.");
+        } else {
+            System.out.println("Now you have " + currentTaskNum + " tasks in the list.");
+        }
+    }
+
+    public int getMarkPosition(String userInput) {
+        words = userInput.split(" ");
+        return Integer.parseInt(words[1]) - 1;
+    }
+    
+    public void toggleMark(int posOfMark, boolean shouldMarkAsDone) {
+        if (!(posOfMark >= 0 && posOfMark <= currentTaskNum)) {
+            handleError();
+        } else {
+            if (shouldMarkAsDone) {
+                list[posOfMark].markAsDone();
+            } else {
+                list[posOfMark].markAsUndone();
+            }
+        }
+    }
+
+    private void processAddTaskRequest() {
+        printAddTaskMessage();
+        currentTaskNum++;
+        printTotalTasks(currentTaskNum);
+        System.out.println(lineBreaker);
+    }
+
+    private static void printAddTaskMessage() {
+        System.out.println(lineBreaker + "Got it. I've added this task:\n"
+                + "  "
+                + list[currentTaskNum].toString()
+                + System.lineSeparator());
+    }
+
+    public void handleRequest(String userInput) {
+        if (userInput.startsWith("mark")) {
+            int posOfMark = getMarkPosition(userInput);
+            toggleMark(posOfMark, true);
+        } else if (userInput.startsWith("unmark")) {
+            int posOfMark = getMarkPosition(userInput);
+            toggleMark(posOfMark, false);
+        } else if (userInput.equals("list")) {
+            System.out.println("Here is your list!");
+            printList(currentTaskNum);
+        } else if (userInput.equals("bye")) {
+            System.out.println(farewellMessage);
+            setShouldExit();
+        } else if (userInput.startsWith("todo")) {
+            list[currentTaskNum] = new ToDos(userInput);
+            processAddTaskRequest();
+        } else if (userInput.startsWith("deadline")) {
+            phrases = userInput.split("/by ");
+            if (phrases.length < 2) {
+                handleError();
+            } else {
+                list[currentTaskNum] = new Deadline(phrases[0], phrases[1]);
+                processAddTaskRequest();
+            }
+        } else if (userInput.startsWith("event")) {
+            phrases = userInput.split("/");
+            if (phrases.length < 3) {
+                handleError();
+            } else {
+                list[currentTaskNum] = new Event(phrases[0], phrases[1], phrases[2]);
+                processAddTaskRequest();
+            }
+        } else {
+            list[currentTaskNum] = new Task(userInput);
+            processAddTaskRequest();
+        }
+    }

```

Comment:
Perhaps you could provide some documentation to summarise the role of this method for better readability